bikram.md

```diff
@@ -0,0 +1,95 @@
+
+13	Enable CORS for all APIs:
+When your API's resources receive requests from a domain other than the API's own domain, you must enable cross-origin resource sharing (CORS) for selected methods on the resource. This amounts to having your API respond to the OPTIONS preflight request with at least the following CORS-required response headers:
+•	Access-Control-Allow-Methods
+•	Access-Control-Allow-Headers
+•	Access-Control-Allow-Origin
+
+14	Data in transit:
+Unless the public information is completely read-only, the use of TLS v1.2 should be mandated, particularly where credentials, updates, deletions, and any value transactions are performed. The overhead of TLS is negligible on modern hardware, with a minor latency increase that is more than compensated by safety for the end user.
+
+15	Message Integrity:
+In addition to HTTPS/TLS, JSON Web Token (JWT) is an open standard that defines a compact and self-contained way for securely transmitting information between parties as a JSON object. JWT can not only be used to ensure the message integrity but also authentication of both message sender/receiver. The JWT includes the digital signature hash value of the message body to ensure the message integrity during the transmission.
+
+16	Weak SSL/TLS Ciphers Support:
+The encryption ciphers supported by the server may allow an attacker to eavesdrop on the connection. Verify the following guidelines:
+•	When serving up content to your users, only strong ciphers are enabled (128 bits and above).
+•	When connecting to other remote systems ensure that your client does not connect using a weak cipher if the server supports it.
+•	Renegotiation must be properly configured (e.g. Insecure Renegotiation must be disabled, due to MiTM attacks and Client-initiated Renegotiation must be disabled, due to Denial of Service vulnerability).
+•	MD5 should not be used, due to known collision attacks.
+•	RC4 should not be used, due to crypto-analytical attacks
+•	Server should be protected from BEAST Attack 
+•	Server should be protected from CRIME attack, TLS compression must be disabled 
+•	Server should support Forward Secrecy
+
+17	Mixed Content:
+When serving up content to your users over SSL, it’s important that you do not include content served over HTTP such as Images, JavaScript, Flash, or CSS. By mixing HTTP content with HTTPS content, you expose your users to man in the middle attacks and eliminate the security benefits that SSL/TLS provides.
+
+18	SSL certificate validity – client and server:
+In order for the communication to be set up, a number of checks on the certificates must be passed:	
+•	Checking if the Certificate Authority (CA) is a known one (meaning one considered trusted);
+•	Checking that the certificate is currently valid;
+•	Checking that the name of the site and the name reported in the certificate match.
+
+19	Certificate Requirements:
+The following checklist needs to be followed while using an SSL certificate:
```

Comment:
@oonovak how are we going to tackle the certificates? are we going to provide them on behalf of the API management?